src/interop.rs

```diff
@@ -64,74 +60,56 @@ impl From<&tex::Person> for Person {
 
 impl From<tex::Date> for Date {
     fn from(date: tex::Date) -> Self {
+        let approximate = date.uncertain || date.approximate;
+
         match date.value {
-            DateValue::At(x) | DateValue::After(x) | DateValue::Before(x) => {
-                Date { year: x.year, month: x.month, day: x.day }
-            }
-            DateValue::Between(_, x) => Self { year: x.year, month: x.month, day: x.day },
+            DateValue::At(x) | DateValue::After(x) | DateValue::Before(x) => Date {
+                year: x.year,
+                month: x.month,
+                day: x.day,
+                approximate,
+            },
+            DateValue::Between(_, x) => Self {
+                year: x.year,
+                month: x.month,
+                day: x.day,
+                approximate,
+            },
         }
     }
 }
 
-impl From<&[Spanned<Chunk>]> for FmtString {
+impl From<&[Spanned<Chunk>]> for ChunkedString {
     fn from(chunks: &[Spanned<Chunk>]) -> Self {
-        let mut fmt = Self::new(chunks.format_verbatim());
-        let verbatim = chunks.iter().all(|chunk| match &chunk.v {
-            Chunk::Normal(s) => s.chars().all(char::is_whitespace),
-            _ => true,
-        });
-
-        fmt = fmt.verbatim(verbatim);
-        if !verbatim {
-            if chunks.iter().any(|chunk| matches!(chunk.v, Chunk::Verbatim(_))) {
-                fmt = fmt.title_case(format_title_case(chunks));
+        let mut res = Self::new();
+        for chunk in chunks {
+            match &chunk.v {
+                Chunk::Normal(s) => res.push_str(s, ChunkKind::Normal),
+                Chunk::Verbatim(s) => res.push_str(s, ChunkKind::Verbatim),
+                Chunk::Math(s) => res.push_str(s, ChunkKind::Math),
             }
-            fmt = fmt.sentence_case(chunks.format_sentence());
         }
-
-        fmt
+        res
     }
 }
-
-fn format_title_case(chunks: &[Spanned<Chunk>]) -> String {
-    let mut out = String::new();
-    for chunk in chunks {
-        match &chunk.v {
-            Chunk::Normal(s) => out.push_str(
-                &TitleCase {
-                    trim_end: false,
-                    trim_start: false,
-                    always_capitalize_last_word: false,
-                    ..Default::default()
-                }
-                .apply(s),
-            ),
-            Chunk::Verbatim(s) => out.push_str(s),
-            Chunk::Math(s) => {
-                out.push('$');
-                out += s;
-                out.push('$');
-            }
-        }
+impl From<&[Spanned<Chunk>]> for FormatString {
+    fn from(chunks: &[Spanned<Chunk>]) -> Self {
+        Self { value: chunks.into(), short: None }
     }
-    out
 }
```

Comment:
This impl feels a bit unidiomatic because it converts _and_ implicitly clones. Is it used at all / a lot?

Reply:
Yes, a lot, for `biblatex` conversion